src/hmm/hmmer.py

```diff
@@ -0,0 +1,477 @@
+        """
+        with HMMFile(hmm_path) as hmm_file:
+            if hmm_file.is_pressed():
+                logging.info("PFAM Hmm file was already pressed.")
+                return
+
+            logging.info("Pressing Pfam HMM file...")
+            hmmpress(hmm_file, hmm_path)
+
+    @staticmethod
+    def init(hmm_path: Path, optimized=True) -> None:
+        """Load the HMMer profiles and create an index for the profiles for easy lookup
+        later. Also creates a pipeline for usage in certain workflows
+
+        Args:
+            hmm_path (Path): Path to the hmm profile. Should contain hmm[f,i,m,p] if
+            optimized is set to True
+            optimized (bool, optional): Whether to use the optimized profiles if they
+            exist. Defaults to True.
+        """
+        logging.info("Reading HMM Profiles")
+        HMMer.profiles = list()
+        with HMMFile(hmm_path) as hmm_file:
+            if optimized:
+                logging.debug("Loading optimized profiles")
+                profile_list = list(hmm_file.optimized_profiles())
+            else:
+                logging.debug("Loading unoptimized profiles")
+                profile_list = list(hmm_file)
+
+            HMMer.profiles.extend(profile_list)
+            logging.info("Found %d HMM profiles", len(HMMer.profiles))
+
+        # create an index so we can quickly get profiles by accession
+        HMMer.profile_index = gen_profile_index(HMMer.profiles)
+
+        # this pipeline is used to perform hmmscan and hmmalign in more memory efficient
+        # use cases
+        HMMer.pipeline = Pipeline(
+            Alphabet.amino(), Z=len(HMMer.profiles), bit_cutoffs="trusted"
+        )
+
+    @staticmethod
+    def unload():
+        """Unload the variables that were set up during init(). Do this before running
+        init() again for hmmalign
+        """
+        logging.info("Unloading profiles")
+        HMMer.profiles = list()
+        HMMer.profile_index = {}
+        HMMer.pipeline = None
+
+    @staticmethod
+    def get_profile(accession: str) -> OptimizedProfile:
+        """Returns the optimized profile with the specified accession
+
+        Args:
+            accession (str): the accession identifying the profile of interest
+
+        Returns:
+            OptimizedProfile: An optimized HMM profile
+        """
+        if accession not in HMMer.profile_index:
+            raise KeyError()
+
+        profile_idx = HMMer.profile_index[accession]
+        return HMMer.profiles[profile_idx]
+
+    @staticmethod
+    def hmmsearch_simple(genes: list[CDS]) -> Iterator[HSP]:
+        """Performs hmmsearch on a list of CDS.
+
+        This is the fastest method of getting scan results. Use this if you know you
+        won't run into memory issues.
+
+        Args:
+            genes (list[CDS]): list of genes to generate HSPs for
+
+        Yields:
+            Iterator[HSP]: an iterator generating HSPs that are included
```

Comment:
I've added the following line, but otherwise its just returning HSP objects

```python
"""
        HSP objects generated by iterator contain domain and score information, and a
        link to the related CDS
"""
```